apps/backend/src/modules/users/user.service.ts

```diff
@@ -52,9 +51,8 @@ class UserService implements Service {
 		};
 	}
 
-	public async findByEmail(email: string): Promise<null | UserDto> {
-		const item = await this.userRepository.findByEmail(email);
-		return item ? item.toObject() : null;
+	public async findByEmail(email: string): Promise<null | UserEntity> {
+		return await this.userRepository.findByEmail(email);
```

Comment:
Return `toObject` call here

Reply:
I cannot convert the result of the findByEmail method to an object using toObject() directly in the repository because this would break the password verification logic in the service. In the service, I need to work with the full UserEntity to access its methods, including toNewObject(), which provides passwordHash and passwordSalt. These fields are necessary for comparing the stored password hash with the password provided by the user. Converting to an object at the repository level would prevent this.

Additionally, using a UserDTO here won’t help because it doesn’t contain the necessary fields for this operation. The service ensures that the response sent back to the client contains only the safe information, excluding the password and salt.
Like:
public async signIn(...){
    return user.toObject();
}